File: ts-sdk/src/sdk/models/operations/putweblinksidnumberaddsharedlink.ts
import { SpeakeasyMetadata, SpeakeasyBase, RetryConfig } from "../../../internal/utils";
import * as shared from "../shared";



export class PutWebLinksIdNumberAddSharedLinkPathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=web_link_id" })
  webLinkId: string;
}


export class PutWebLinksIdNumberAddSharedLinkQueryParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=fields" })
  fields: string;
}

export enum PutWebLinksIdNumberAddSharedLinkRequestBodySharedLinkAccessEnum {
    Open = "open",
    Company = "company",
    Collaborators = "collaborators"
}


export class PutWebLinksIdNumberAddSharedLinkRequestBodySharedLinkPermissions extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=can_download" })
  canDownload?: boolean;

  @SpeakeasyMetadata({ data: "json, name=can_edit" })
  canEdit?: boolean;

  @SpeakeasyMetadata({ data: "json, name=can_preview" })
  canPreview?: boolean;
}


// PutWebLinksIdNumberAddSharedLinkRequestBodySharedLink
/** 
 * The settings for the shared link to create on the web link.
 * 
 * Use an empty object (`{}`) to use the default settings for shared
 * links.
**/
export class PutWebLinksIdNumberAddSharedLinkRequestBodySharedLink extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=access" })
  access?: PutWebLinksIdNumberAddSharedLinkRequestBodySharedLinkAccessEnum;

  @SpeakeasyMetadata({ data: "json, name=password" })
  password?: string;

  @SpeakeasyMetadata({ data: "json, name=permissions" })
  permissions?: PutWebLinksIdNumberAddSharedLinkRequestBodySharedLinkPermissions;

  @SpeakeasyMetadata({ data: "json, name=unshared_at" })
  unsharedAt?: Date;

  @SpeakeasyMetadata({ data: "json, name=vanity_name" })
  vanityName?: string;
}


export class PutWebLinksIdNumberAddSharedLinkRequestBody extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=shared_link" })
  sharedLink?: PutWebLinksIdNumberAddSharedLinkRequestBodySharedLink;
}


export class PutWebLinksIdNumberAddSharedLinkRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  retries?: RetryConfig;

  @SpeakeasyMetadata()
  pathParams: PutWebLinksIdNumberAddSharedLinkPathParams;

  @SpeakeasyMetadata()
  queryParams: PutWebLinksIdNumberAddSharedLinkQueryParams;

  @SpeakeasyMetadata({ data: "request, media_type=application/json" })
  request?: PutWebLinksIdNumberAddSharedLinkRequestBody;
}


export class PutWebLinksIdNumberAddSharedLinkResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  clientError?: shared.ClientError;

  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;

  @SpeakeasyMetadata()
  webLink?: shared.WebLink;
}

File: sdks/go/pkg/models/operations/putweblinksidnumberaddsharedlink.go
package operations

import (
	"openapi/pkg/models/shared"
	"openapi/pkg/utils"
	"time"
)

type PutWebLinksIDNumberAddSharedLinkPathParams struct {
	WebLinkID string `pathParam:"style=simple,explode=false,name=web_link_id"`
}

type PutWebLinksIDNumberAddSharedLinkQueryParams struct {
	Fields string `queryParam:"style=form,explode=true,name=fields"`
}

type PutWebLinksIDNumberAddSharedLinkRequestBodySharedLinkAccessEnum string

const (
	PutWebLinksIDNumberAddSharedLinkRequestBodySharedLinkAccessEnumOpen          PutWebLinksIDNumberAddSharedLinkRequestBodySharedLinkAccessEnum = "open"
	PutWebLinksIDNumberAddSharedLinkRequestBodySharedLinkAccessEnumCompany       PutWebLinksIDNumberAddSharedLinkRequestBodySharedLinkAccessEnum = "company"
	PutWebLinksIDNumberAddSharedLinkRequestBodySharedLinkAccessEnumCollaborators PutWebLinksIDNumberAddSharedLinkRequestBodySharedLinkAccessEnum = "collaborators"
)

type PutWebLinksIDNumberAddSharedLinkRequestBodySharedLinkPermissions struct {
	CanDownload *bool `json:"can_download,omitempty"`
	CanEdit     *bool `json:"can_edit,omitempty"`
	CanPreview  *bool `json:"can_preview,omitempty"`
}

// PutWebLinksIDNumberAddSharedLinkRequestBodySharedLink
// The settings for the shared link to create on the web link.
//
// Use an empty object (`{}`) to use the default settings for shared
// links.
type PutWebLinksIDNumberAddSharedLinkRequestBodySharedLink struct {
	Access      *PutWebLinksIDNumberAddSharedLinkRequestBodySharedLinkAccessEnum  `json:"access,omitempty"`
	Password    *string                                                           `json:"password,omitempty"`
	Permissions *PutWebLinksIDNumberAddSharedLinkRequestBodySharedLinkPermissions `json:"permissions,omitempty"`
	UnsharedAt  *time.Time                                                        `json:"unshared_at,omitempty"`
	VanityName  *string                                                           `json:"vanity_name,omitempty"`
}

type PutWebLinksIDNumberAddSharedLinkRequestBody struct {
	SharedLink *PutWebLinksIDNumberAddSharedLinkRequestBodySharedLink `json:"shared_link,omitempty"`
}

type PutWebLinksIDNumberAddSharedLinkRequest struct {
	Retries     *utils.RetryConfig
	PathParams  PutWebLinksIDNumberAddSharedLinkPathParams
	QueryParams PutWebLinksIDNumberAddSharedLinkQueryParams
	Request     *PutWebLinksIDNumberAddSharedLinkRequestBody `request:"mediaType=application/json"`
}

type PutWebLinksIDNumberAddSharedLinkResponse struct {
	ClientError *shared.ClientError
	ContentType string
	StatusCode  int64
	WebLink     *shared.WebLink
}
